relayer/processor/path_processor.go

```diff
@@ -61,25 +71,58 @@ func NewPathProcessor(log *zap.Logger, pathEnd1 PathEnd, pathEnd2 PathEnd) *Path
 		pathEnd1: &PathEndRuntime{
 			info:              pathEnd1,
 			incomingCacheData: make(chan ChainProcessorCacheData, 100),
-			messageCache:      make(MessageCache),
+			channelStateCache: make(ChannelStateCache),
+			messageCache:      make(ChannelMessageCache),
 		},
 		pathEnd2: &PathEndRuntime{
 			info:              pathEnd2,
 			incomingCacheData: make(chan ChainProcessorCacheData, 100),
-			messageCache:      make(MessageCache),
+			channelStateCache: make(ChannelStateCache),
+			messageCache:      make(ChannelMessageCache),
 		},
 		retryProcess: make(chan struct{}, 8),
 	}
 }
 
 // TEST USE ONLY
-func (pp *PathProcessor) PathEnd1Messages(message string) SequenceCache {
-	return pp.pathEnd1.messageCache[message]
+func (pp *PathProcessor) PathEnd1Messages(channelKey ChannelKey, message string) SequenceCache {
+	return pp.pathEnd1.messageCache[channelKey][message]
 }
 
 // TEST USE ONLY
-func (pp *PathProcessor) PathEnd2Messages(message string) SequenceCache {
-	return pp.pathEnd2.messageCache[message]
+func (pp *PathProcessor) PathEnd2Messages(channelKey ChannelKey, message string) SequenceCache {
+	return pp.pathEnd2.messageCache[channelKey][message]
+}
+
+type channelPair struct {
+	pathEnd1ChannelKey ChannelKey
+	pathEnd2ChannelKey ChannelKey
+}
+
+func (pp *PathProcessor) channelPairs() []channelPair {
+	channelsFromPathEnd1Perspective := make(map[ChannelKey]bool)
```

Comment:
This is another unusually long variable name. Maybe just call it `channels` with a comment mentioning the channels are from path end 1's perspective?

Reply:
done